File: LeetCode/66. Plus One/Solution.java
public class Solution {
    public int[] plusOne(int[] digits) {
        int i = digits.length - 1;
        while (i >= 0) {
            if (digits[i] == 9) {
                digits[i] = 0;
                i--;
            } else {
                digits[i]++;
                break;
            }
        }
        if (i == -1) {//if digits is 9999, then return 10000
            int[] nums = new int[digits.length + 1];
            System.arraycopy(digits, 0, nums, 1, digits.length);
            nums[0] = 1;
            return nums;
        }
        return digits;
    }
}

File: LeetCode/66. Plus One/solution.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        i = len(digits) - 1
        digits[i] += 1
        while digits[i] == 10:
            digits[i] = 0
            i -= 1
            if i >= 0:
                digits[i] += 1
            else:
                break
        if digits[0] == 0:
            digits.insert(0, 1)
        return digits

    def plusOne1(self, digits):
        num = 0
        for i in digits:
            num = num * 10 + i
        return [int(d) for d in str(num + 1)]
